pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
 	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
 
+	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
+		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
```

Comment:
I don't like calling docker before EnsureDocker - if ensure docker is
called from build* arguments that's acceptable.

On Feb 1, 2016, at 4:54 PM, Avesh Agarwal <notifications@github.com> wrote:

In pkg/cmd/server/kubernetes/node_config.go
<https://github.com/openshift/origin/pull/6538#discussion_r51487255>:

> @@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
>  	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
>  	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
>
> +	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
> +		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {

SerializeImagePulls is bool, and it wouldn't be possible to know if user
changed it or not where Ensuredocker is called, that would mean mutating
user configured value of SerializeImagePulls.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/6538/files#r51487255>.